rtpengine: fails with kernel 5.4.78+

@micmac1 @neheb 

Bumping kernel to 5.4.78/79 (in preparation) breaks build of the rtpengine package (full errors at the end).

This appears to be caused by the following kernel commit, adding an argument to the ip_route_me_harder function:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable-rc.git/commit/?h=queue/5.4&id=b255b5cab13ac78c43b8c10c0e904b3e6f76b251

An equivalent issue has already been addressed for xtables-addons in packages repo:
https://github.com/openwrt/packages/issues/13947

The kernel bump itself is tracked here:
https://github.com/openwrt/openwrt/pull/3604

Errors:
```
/scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module/xt_RTPENGINE.c: In function 'send_proxy_packet4':
/scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module/xt_RTPENGINE.c:3355:42: error: passing argument 2 of 'ip_route_me_harder' from incompatible pointer type [-Werror=incompatible-pointer-types]
  if (ip_route_me_harder(par->state->net, skb, RTN_UNSPEC))
                                          ^~~
In file included from ./include/uapi/linux/netfilter_ipv4/ip_tables.h:22,
                 from ./include/linux/netfilter_ipv4/ip_tables.h:23,
                 from /scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module/xt_RTPENGINE.c:25:
./include/linux/netfilter_ipv4.h:19:54: note: expected 'struct sock *' but argument is of type 'struct sk_buff *'
 int ip_route_me_harder(struct net *net, struct sock *sk, struct sk_buff *skb, unsigned addr_type);
                                         ~~~~~~~~~~~~~^~
/scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module/xt_RTPENGINE.c:3355:6: error: too few arguments to function 'ip_route_me_harder'
  if (ip_route_me_harder(par->state->net, skb, RTN_UNSPEC))
      ^~~~~~~~~~~~~~~~~~
In file included from ./include/uapi/linux/netfilter_ipv4/ip_tables.h:22,
                 from ./include/linux/netfilter_ipv4/ip_tables.h:23,
                 from /scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module/xt_RTPENGINE.c:25:
./include/linux/netfilter_ipv4.h:19:5: note: declared here
 int ip_route_me_harder(struct net *net, struct sock *sk, struct sk_buff *skb, unsigned addr_type);
     ^~~~~~~~~~~~~~~~~~
/scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module/xt_RTPENGINE.c: In function 'send_proxy_packet6':
/scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module/xt_RTPENGINE.c:3451:43: error: passing argument 2 of 'ip6_route_me_harder' from incompatible pointer type [-Werror=incompatible-pointer-types]
  if (ip6_route_me_harder(par->state->net, skb))
                                           ^~~
In file included from /scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module/xt_RTPENGINE.c:27:
./include/linux/netfilter_ipv6.h:146:55: note: expected 'struct sock *' but argument is of type 'struct sk_buff *'
 int ip6_route_me_harder(struct net *net, struct sock *sk, struct sk_buff *skb);
                                          ~~~~~~~~~~~~~^~
/scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module/xt_RTPENGINE.c:3451:6: error: too few arguments to function 'ip6_route_me_harder'
  if (ip6_route_me_harder(par->state->net, skb))
      ^~~~~~~~~~~~~~~~~~~
In file included from /scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module/xt_RTPENGINE.c:27:
./include/linux/netfilter_ipv6.h:146:5: note: declared here
 int ip6_route_me_harder(struct net *net, struct sock *sk, struct sk_buff *skb);
     ^~~~~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[6]: *** [scripts/Makefile.build:266: /scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module/xt_RTPENGINE.o] Error 1
make[5]: *** [Makefile:1729: /scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module] Error 2
make[5]: Leaving directory '/scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/linux-5.4.79'
make[4]: *** [Makefile:34: modules] Error 2
make[4]: Leaving directory '/scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/kernel-module'
make[3]: *** [Makefile:267: /scratch/union/build_dir/target-mips_24kc_musl/linux-ath79_generic/rtpengine-no-transcode/rtpengine-mr8.5.2.1/.built] Error 2
make[3]: Leaving directory '/scratch/union/feeds/telephony/net/rtpengine'
time: package/feeds/telephony/rtpengine/no-transcode/compile#1.39#0.39#3.39
make[2]: *** [package/Makefile:114: package/feeds/telephony/rtpengine/compile] Error 2
make[2]: *** Waiting for unfinished jobs....
```

--- a/net/rtpengine/Makefile
+++ b/net/rtpengine/Makefile
@@ -9,12 +9,12 @@ include $(TOPDIR)/rules.mk
 include $(INCLUDE_DIR)/kernel.mk
 
 PKG_NAME:=rtpengine
-PKG_VERSION:=mr8.5.2.1
+PKG_VERSION:=mr8.5.2.3
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://codeload.github.com/sipwise/rtpengine/tar.gz/$(PKG_VERSION)?
-PKG_HASH:=7c6f0e036d9aa29485236cc9cd7157625071038f5d773c54c6dc0dc4606aed1e
+PKG_HASH:=e2c0f94984753f321e1db0a97f81af18f3e9f7005a2ca67766ddef2b6c8d4427
 
 PKG_LICENSE:=GPL-3.0
 PKG_LICENSE_FILES:=LICENSE

--- a/net/rtpengine/patches/05-support-kernels-greater-equal-5.9.9.patch
+++ b/net/rtpengine/patches/05-support-kernels-greater-equal-5.9.9.patch
@@ -1,0 +1,38 @@
+commit 46a64889071cb844550786096c251b804a234016
+Author: Richard Fuchs <rfuchs@sipwise.com>
+Date:   Thu Nov 19 11:22:26 2020 -0500
+
+    support kernels >= 5.9.9
+    
+    closes #1111
+    
+    Change-Id: I81f22bfec93b38a108b671b10bd70bf86fb1270c
+
+diff --git a/kernel-module/xt_RTPENGINE.c b/kernel-module/xt_RTPENGINE.c
+index c12edcd8..3c5e3a89 100644
+--- a/kernel-module/xt_RTPENGINE.c
++++ b/kernel-module/xt_RTPENGINE.c
+@@ -3351,7 +3351,10 @@ static int send_proxy_packet4(struct sk_buff *skb, struct re_address *src, struc
+ 		uh->check = CSUM_MANGLED_0;
+ 
+ 	skb->protocol = htons(ETH_P_IP);
+-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,10,0)
++#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,9,9) || \
++		(LINUX_VERSION_CODE >= KERNEL_VERSION(5,4,78) && LINUX_VERSION_CODE < KERNEL_VERSION(5,5,0))
++	if (ip_route_me_harder(par->state->net, par->state->sk, skb, RTN_UNSPEC))
++#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4,10,0)
+ 	if (ip_route_me_harder(par->state->net, skb, RTN_UNSPEC))
+ #elif LINUX_VERSION_CODE >= KERNEL_VERSION(4,4,0)
+ 	if (ip_route_me_harder(par->net, skb, RTN_UNSPEC))
+@@ -3447,7 +3450,10 @@ static int send_proxy_packet6(struct sk_buff *skb, struct re_address *src, struc
+ 		uh->check = CSUM_MANGLED_0;
+ 
+ 	skb->protocol = htons(ETH_P_IPV6);
+-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,10,0)
++#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,9,9) || \
++		(LINUX_VERSION_CODE >= KERNEL_VERSION(5,4,78) && LINUX_VERSION_CODE < KERNEL_VERSION(5,5,0))
++	if (ip6_route_me_harder(par->state->net, par->state->sk, skb))
++#elif LINUX_VERSION_CODE >= KERNEL_VERSION(4,10,0)
+ 	if (ip6_route_me_harder(par->state->net, skb))
+ #elif LINUX_VERSION_CODE >= KERNEL_VERSION(4,4,0)
+ 	if (ip6_route_me_harder(par->net, skb))

